.travis.yml

```diff
@@ -91,160 +53,150 @@ matrix:
-               MATPLOTLIB_VERSION="<3.1"
-               PIP_DEPENDENCIES="" CONDA_DEPENDENCIES=""
-               INSTALL_WITH_PIP=True
-               EXTRAS_INSTALL="docs"
-
-        # Testing with Travis provided Python3.8. Change this job to a regular
-        # one once conda supports python3.8. Make sure we use released numpy rather
-        # than some kind of dev coming natively from the Travis environment
+          env: TOXENV='linkcheck'
+               EVENT_TYPE='cron'
+          addons:
+              apt:
+                  packages:
+                      - graphviz
+
+        # Testing with Travis provided Python3.8.
         - language: python
           stage: Final tests
           dist: xenial
-          python: 3.8-dev
-          env: INSTALL_WITH_PIP=True EXTRAS_INSTALL="test" USE_CI_HELPERS=False
-               OTHER_DEPENDENCY_PREFERENCES='numpy==1.17.3' PIP_FLAGS="--pre"
+          python: 3.8
 
     allow_failures:
-      - os: linux
-        stage: Final tests
-        env: PYTHON_VERSION=3.7 NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-             CONDA_DEPENDENCIES=''
-             PIP_DEPENDENCIES=$DEV_PIP_DEP
-             MATPLOTLIB_VERSION=dev
+        - language: python
+          python: 3.7
+          stage: Final tests
+          env: TOXENV="test-devdeps"
+               TOXPOSARGS="--remote-data"
 
 before_install:
 
+    # We need a full clone to make sure setuptools_scm works properly
+    - git fetch --unshallow .
+    - git fetch --depth=1000000
+
     # We need to use CCOMPILER otherwise Travis overwrites CC if we define it
     # in env: above.
     - if [ ! -z $CCOMPILER ]; then
         export CC=$CCOMPILER;
       fi
 
      # Check CC variable
-    - echo $CC
+    - echo "CC="$CC
 
     # Write configuration items to standard location to make sure they are
     # ignored (the tests will fail if not)
     - mkdir -p $HOME/.astropy/config/
     - printf "unicode_output = True\nmax_width = 500" > $HOME/.astropy/config/astropy.cfg
 
+    - if [[ $COVERAGE == True ]]; then
+        export TOXPOSARGS=$TOXPOSARGS" --cov-report=xml:"$TRAVIS_BUILD_DIR"/coverage.xml";
+      fi
+
+    - echo "TOXENG="$TOXENG
+    - echo "TOXARGS="$TOXARGS
+    - echo "TOXPOSARGS="$TOXPOSARGS
+
+
 
 install:
     - if [[ $USE_CI_HELPERS == True ]]; then
         git clone git://github.com/astropy/ci-helpers.git;
         source ci-helpers/travis/setup_conda.sh;
       fi
-    - if [[ $INSTALL_WITH_PIP == True ]]; then
-        if [ -z $EXTRAS_INSTALL ]; then
-          pip install -e . ${OTHER_DEPENDENCY_PREFERENCES} ${PIP_FLAGS};
-        else
-          pip install -e .[$EXTRAS_INSTALL] ${OTHER_DEPENDENCY_PREFERENCES} ${PIP_FLAGS};
-        fi
-      fi
 
 script:
-    - if [[ $DOUBLE_PLAY == True ]]; then
-          $MAIN_CMD $SETUP_CMD;
-          $MAIN_CMD $SETUP_CMD;
-      else
-          $MAIN_CMD $SETUP_CMD;
-      fi
+    - pip install tox
+    - tox -e $TOXENV $TOXARGS -- $TOXPOSARGS
```

Comment:
`-e $TOXENV` is probably useless here since tox will read the env var by itself.

Reply:
Fixed